code/modules/admin/smites/imaginary_friend_special.dm

```diff
@@ -47,18 +78,33 @@
 		to_chat(user, span_warning("The target mob no longer exists, aborting."))
 		return
 
-	if(QDELETED(friend_candidate_client))
-		to_chat(user, span_warning("Imaginary friend candidate no longer has a client, aborting."))
+	if(!length(friend_candidates))
+		to_chat(user, span_warning("No provided imaginary friend candidates, aborting."))
+		return
+
+	var/list/final_clients = list()
+	for (var/client/client as anything in friend_candidates)
+		if (QDELETED(client))
+			continue
+		final_clients += client
+
+	if(!length(final_clients))
+		to_chat(user, span_warning("No provided imaginary friend candidates had clients, aborting."))
 		return
 
-	if(isliving(friend_candidate_client.mob))
-		friend_candidate_client.mob.ghostize(can_reenter_corpse = TRUE)
+	for (var/client/friend_candidate_client as anything in final_clients)
+		if(isliving(friend_candidate_client.mob))
+			friend_candidate_client.mob.ghostize(can_reenter_corpse = TRUE)
+
+		var/mob/camera/imaginary_friend/friend_mob
 
-	var/mob/camera/imaginary_friend/friend_mob
+		if(random_appearance || isnull(friend_candidate_client.prefs))
+			friend_mob = new /mob/camera/imaginary_friend(get_turf(target), target)
+		else
+			friend_mob = new /mob/camera/imaginary_friend(get_turf(target), target, friend_candidate_client.prefs)
 
-	if(random_appearance)
-		friend_mob = new /mob/camera/imaginary_friend(get_turf(target), target)
-	else
-		friend_mob = new /mob/camera/imaginary_friend(get_turf(target), target, friend_candidate_client.prefs)
+		friend_mob.key = friend_candidate_client.key
```

Comment:
i'm not fond of the pattern of doing all of this and then manually changing the keys over to have this work, it feels janky and too break-prone. can you retool this to use something like `change_mob_type()` instead? we already interact with the mob associated with the client we pass in so just cache that to a variable in this for loop and do stuff with that instead